### DISCUSSION TITLE
How to stitch two images vertically with same width and different height?

### DISCUSSION THREAD

**Discussion by rebornfz** [libvips/libvips#3195] [Q&A] [answered]
I want to stitch two images vertically, they have the same widht and different height.
example:
pic1
![1](https://user-images.githubusercontent.com/27042311/204212979-8dd699b5-7e10-4dd8-b187-5c5c7ac31033.png)
pic2
![2](https://user-images.githubusercontent.com/27042311/204213021-3b42f612-1023-4988-951c-38e0eeb3b91a.png)
I use arrayjoin get the result
![image_out1](https://user-images.githubusercontent.com/27042311/204213139-bdf0b914-d024-4bc4-9af6-c27d6cc10bc3.png)
The sample code:
```
#include <vips/vips8>
#include <vector>
#include <iostream>
#include <chrono>


int main(int argc, char **argv)
{
    VIPS_INIT (argv[0]);

    vips::VImage png1 = vips::VImage::new_from_file("1.png");
    vips::VImage png2 = vips::VImage::new_from_file("2.png");

    std::vector<vips::VImage> images;
    std::vector<int> delay_array;
    for(int i = 0; i < 2; i++){
        delay_array.push_back(1000);
    }
    images.push_back(png1);
    images.push_back(png2);

    vips::VImage animation = vips::VImage::arrayjoin(images, vips::VImage::option ()->set ("across", 1));
    animation.write_to_file ("image_out1.png");
}
```

The expected result
![test](https://user-images.githubusercontent.com/27042311/204213199-d0ab2c6d-b7e9-4f87-99bc-e9676254448c.png)
How can I get the expected result

**Comment by jcupitt** [comment, marked as answer]
Hi, you need:

```
vips join top.png bottom.png out.png vertical
```

**Reply by jcupitt** [reply]
Or 

```C++
VImage out = png1.join(png2, VIPS_DIRECTION_VERTICAL);
```
